assignment_2.java

```diff
@@ -0,0 +1,168 @@
+        String Repeated = "";
+        for (int i = 0; i < length; i++) {
+            count =0;
+            for (int j = i; j < length; j++) {
+                if (Words.charAt(i) == Words.charAt(j)) {
+                    count++;
+                }
+                    Repeated = String.valueOf(Words.charAt(i));
+            }
+        if(MaxCount<=count)
+        {
+            MaxCount=count;
+            Repeated += String.valueOf(Words.charAt(i));
+        }
+        }
+        System.out.print(Repeated + " is the max repeated word " + MaxCount + " times");
+    }
+
+
+
+
+
+
+
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+        option();
+        int num = input.nextInt();
+        input.nextLine();
+        switch (num) {
+            case 1:
+                System.out.print(code_word.Append_ENTER_STRING);
+                String current_string = input.next();
+                do {
+                    System.out.print(code_word.Append_ENTER_SUBSTRING);
+                    String new_string = input.next();
+                    current_string = append_string(current_string,new_string);
+                    System.out.println(code_word.Append_UPDATED_STRING + current_string);
+                    System.out.println(code_word.Append_PROMPT_YES_NO);
+                } while (input.next().equals("Y".toLowerCase()));
+                break;
+
+
+            case 2:
+                System.out.print("Enter the String for counting words: ");
+                String string = input.nextLine();
+                System.out.print("Word Count is: " + count_words(string));
+                break;
+
+
+            case 3:
+                System.out.print("Enter the String: ");
+                String From_user = input.nextLine();
+                System.out.print("Enter the String or Char you want to replace: ");
+                String Old_string = input.nextLine();
+                System.out.print("Enter the String or Char you want to replace: ");
+                String New_string = input.nextLine();
+                Replace_char(From_user,Old_string,New_string);
+                break;
+
+
+            case 4:
+                System.out.print("Enter the String: ");
+                String Reverse_string = input.next();
+                String Reversed_string = Reverse(Reverse_string);
+                System.out.print("Reversed String is: " + Reversed_string);
+                break;
+
+
+            case 5:
+                System.out.print("Enter the String to check Palindrome: ");
+                String Palindrome_string = input.next();
+                String check = CheckPalindrome(Palindrome_string);
+                System.out.print(check);
+                break;
+
+
+            case 6:
+                System.out.print("Enter the String: ");
+                String User_string = input.next();
+                MaxRepeat(User_string);
+                break;
+
+
+
+

```

Comment:
Missing break statements after some case blocks in the switch statement.

No exception handling in the code.

 Methods are missing comments and explanations.

Inefficient string operations (e.g., concatenation in loops).

No unit tests for methods.